crates/broker/src/errors.rs

```diff
@@ -5,3 +5,31 @@
 pub trait CodedError: std::error::Error {
     fn code(&self) -> &str;
 }
+
+// Macro for implementing Debug for CodedError. Ensures the error code is included in the debug output.
+#[macro_export]
+macro_rules! impl_coded_debug {
+    ($name:ident) => {
```

Comment:
It's unclear to me that this needs to be a macro rather than an blanket impl of the trait. I can try our a change here.

Reply:
Still getting the hang of Rust's type system, but it didn't seem possible to do something like:

```
impl<T: CodedError> Debug for T {
    fn fmt(&self, f: &mut Formatter) -> fmt::Result { ... }
}
```

since a 'foreign' trait for generic type isn't allowed. And can't do `impl Debug for CodedError` since CodedError is a trait not a type